xdsl/dialects/linalg.py

```diff
@@ -534,6 +530,30 @@ def print(self, printer: Printer):
                 )
                 printer.print(")")
 
+    @staticmethod
+    def body_arg_types(
+        operands: Sequence[SSAValue],
+    ) -> Sequence[AnyFloat | IntegerType]:
+        """
+        Return the element types of the arguments of the body of this operation
+        """
+
+        result: Sequence[AnyFloat | IntegerType] = []
+
+        for op_type in (op.type for op in operands):
+            if isa(op_type, MemRefType[Attribute]):
+                element_type = op_type.get_element_type()
+            elif isa(op_type, TensorType[Attribute]):
+                element_type = op_type.get_element_type()
+            elif isinstance(op_type, Operation):
+                element_type = op_type.results[0].type
```

Comment:
Since `Operation` does not inherit from `Attribute` this case should be invalid imo.

Reply:
Indeed! This was added because of a failing test that still managed to pass an operation as an attribute, but I now just fixed the test properly instead.